### ISSUE TITLE
symbol collision with libcrypto.a and providers/liblegacy.a with "no-shared"

### ISSUE THREAD

**Issue by geoff-nixon** [openssl/openssl#13627]
Platform: OS X Catalina); Version: Bloody; Tooling: Standard

When I try to build 3.0.0aX w/o dynamic libraries, several tests fail as below. If I manually `ar x` providers/liblegacy.a, delete liblegacy-lib-x86_64cpuid.o, and then wrap the archive up again, we're good to go.

```
    $ perl ./Configure no-shared
       make
       ...
    ${LDCMD:-clang} -arch x86_64 -Os -arch x86_64 -mtune=native -L. -Wl,-search_paths_first \
		-o fuzz/client-test \
		fuzz/client-test-bin-client.o \
		fuzz/client-test-bin-test-corpus.o \
		-lssl -lcrypto
duplicate symbol '_OPENSSL_wipe_cpu' in:
    providers/liblegacy.a(liblegacy-lib-x86_64cpuid.o)
    ./libcrypto.a(libcrypto-lib-x86_64cpuid.o)
duplicate symbol '_OPENSSL_instrument_bus' in:
    providers/liblegacy.a(liblegacy-lib-x86_64cpuid.o)
    ./libcrypto.a(libcrypto-lib-x86_64cpuid.o)
duplicate symbol '_OPENSSL_ia32_rdrand_bytes' in:
    providers/liblegacy.a(liblegacy-lib-x86_64cpuid.o)
    ./libcrypto.a(libcrypto-lib-x86_64cpuid.o)
duplicate symbol '_OPENSSL_ia32_rdseed_bytes' in:
    providers/liblegacy.a(liblegacy-lib-x86_64cpuid.o)
    ./libcrypto.a(libcrypto-lib-x86_64cpuid.o)
duplicate symbol '_CRYPTO_memcmp' in:
    providers/liblegacy.a(liblegacy-lib-x86_64cpuid.o)
    ./libcrypto.a(libcrypto-lib-x86_64cpuid.o)
duplicate symbol '_OPENSSL_cleanse' in:
    providers/liblegacy.a(liblegacy-lib-x86_64cpuid.o)
    ./libcrypto.a(libcrypto-lib-x86_64cpuid.o)
duplicate symbol '_OPENSSL_ia32_cpuid' in:
    providers/liblegacy.a(liblegacy-lib-x86_64cpuid.o)
    ./libcrypto.a(libcrypto-lib-x86_64cpuid.o)
duplicate symbol '_OPENSSL_atomic_add' in:
    providers/liblegacy.a(liblegacy-lib-x86_64cpuid.o)
    ./libcrypto.a(libcrypto-lib-x86_64cpuid.o)
duplicate symbol '_OPENSSL_rdtsc' in:
    providers/liblegacy.a(liblegacy-lib-x86_64cpuid.o)
    ./libcrypto.a(libcrypto-lib-x86_64cpuid.o)
    duplicate symbol '_OPENSSL_instrument_bus2' in:
    providers/liblegacy.a(liblegacy-lib-x86_64cpuid.o)
    ./libcrypto.a(libcrypto-lib-x86_64cpuid.o)
ld: 10 duplicate symbols for architecture x86_64
```

**Comment by t8m** [comment]
#11311 is related

**Comment by levitte** [comment]
I'm unable to reproduce this on Linux, is it possible to do?

**Comment by levitte** [comment]
We closed #11311 because it doesn't seem to be current any more, so closing this.  Should this problem reappear, please raise a new issue.
